inst/editor/src/Shiny-Ui-Elements/nodeInfoFactory.tsx

```diff
@@ -22,66 +26,107 @@ import type { unknownUiFunctionInfo } from "./UnknownUiFunction";
  * @returns Function to build info for a ui node that has the arguments provided
  * by the `Args` parameter.
  */
-export function nodeInfoFactory<Args extends UiArgumentsObject>() {
+export function nodeInfoFactory<Args extends namedArgsObject>() {
   return function makeInfo<
-    Name extends string,
+    RFnName extends string,
     TakesChildren extends boolean,
     Comp extends UiNodeComponent<Args, { TakesChildren: TakesChildren }>,
-    Lib extends string = "Internal",
+    RPackage extends string = "none",
+    ID extends string = RFnName,
+    PyPackage extends string = "none",
     Cat extends string = "Uncategorized"
   >({
-    name,
-    library,
+    id,
+    r_fn_name,
+    r_package,
+    py_package,
     category,
     ...info
-  }: {
-    library?: Lib;
-    category?: Cat;
-  } & CommonInfo<Args, Name, Comp, TakesChildren>) {
+  }: CommonInfo<
+    Args,
+    ID,
+    Cat,
+    RFnName,
+    RPackage,
+    PyPackage,
+    TakesChildren,
+    Comp
+  >) {
     return {
-      uiName: library ? `${library}::${name}` : name,
-      name,
-      library,
+      id: id ?? r_fn_name,
+      r_fn_name,
+      r_package: r_package ?? "none",
+      py_package: py_package ?? "none",
       category: category ?? "Uncategorized",
       ...info,
-    } as InfoOut<Args, Name, Comp, Lib, Cat, TakesChildren>;
```

Comment:
Good summary of the changes to the info object. 

- `uiName` has been split into `id` for uniquely identifying a node and `r_fn_name` for identifying how to address that function in R code. Currently, `id` will default to `r_fn_name`. Old `name` argument is gone. 
- `library` split into `r_package` and `py_package` to reflect name of the package the function the info object is describing is located.
-